### DISCUSSION TITLE
Changing other players avatar?

### DISCUSSION THREAD

**Discussion by tomasrivero** [wxyz-abcd/node-haxball#60] [Q&A]
How to do so? On the official api this is achieved with room.setPlayerAvatar({avatar} , {playerId})
![image](https://user-images.githubusercontent.com/31349354/223524778-de18ce8c-5732-4349-a573-916f5b9e2493.png)

**Comment by wxyz-abcd** [comment]
You should be able to use the API function: `room.setPlayerAvatar(id, value, headless)` where id is the player's id, value is the avatar, and headless is a boolean that states whether you are trying to change the normal avatar or the headless avatar. However; I'm not certain why it says that room.setPlayerAvatar is not a function, since it is a function.

**Comment by wxyz-abcd** [comment]
I've just created a simple test and it worked correctly, here it is:
```js
room.onPlayerJoin = (playerObj, customData) => {
  room.setPlayerAvatar(playerObj.V, "XY", false);
};
```
Maybe you are using an outdated version. I hope that I'm updating npm version this weekend or next week, in case you have to use the npm package.

**Reply by tomasrivero** [reply]
I'm using the CDN version right now, is that one up to date?

**Reply by wxyz-abcd** [reply]
Yes, but not exactly. It follows the Github files from 1 or 2 days behind. But that should already be more than enough for that code to run correctly.

Maybe you should write part of your code here so that I might be able to fix it.
- Are you on browser or node.js?
- How did you initialize the API?
- Which method are you using to develop your bot?
